<filter> with placeholders in Dockerfile is broken

### Description
<!--
Thank you for trying out Eclipse JKube!

Please fill out the following information to help us in analyzing the issue, but feel free to skip it if you don't have the information at hand or if it does not apply. Please remove every section which does not apply to the issue.
-->

### Info

* Eclipse JKube version : 1.0.2
* Maven version (`mvn -v`) :
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /usr/local/apache-maven-3.6.3
Java version: 11.0.7, vendor: AdoptOpenJDK, runtime: /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
Default locale: en_GB, platform encoding: UTF-8
OS name: "mac os x", version: "10.15.7", arch: "x86_64", family: "mac"
```
* Kubernetes / Red Hat OpenShift setup and version :
```
oc v3.11.0+0cbc58b
kubernetes v1.11.0+d4cacc0
```

* If it's a bug, how to reproduce :
It should be possible to use maven properties denoted by ${propertyname} within a Dockerfile (default \<filter\> feature)
Default configuration element\<filter\> delimiter is used, ie. ${*} 

For example, in maven pom.xml:
```
	<properties>
		<rev.rhel-atomic>7.8-367</rev.rhel-atomic>
		<project.base-image.uri>registry.access.redhat.com/rhel7-atomic:${rev.rhel-atomic}</project.base-image.uri>
```

In Dockerfile:
```
FROM ${project.base-image.uri} AS BUILDER
```

This works just fine with fabric8-maven-plugin;  Having switched to JKube, the filtering is clearly not happening:
```
[INFO] --- kubernetes-maven-plugin:1.0.2:build (image-build) @ myproject-builder ---
[INFO] k8s: Running in Kubernetes mode
[INFO] k8s: Building Docker image in Kubernetes mode
[INFO] k8s: Generators:
[INFO] k8s:  - quarkus
[INFO] k8s:  - spring-boot
[INFO] k8s:  - wildfly-swarm
[INFO] k8s:  - thorntail-v2
[INFO] k8s:  - wildfly-jar
[INFO] k8s:  - openliberty
[INFO] k8s:  - karaf
[INFO] k8s:  - vertx
[INFO] k8s:  - java-exec
[INFO] k8s:  - webapp
[ERROR] k8s: Failed to execute the build [Error while trying to build the image: Unable to check image [${project.base-image.uri}] : {"message":"no such image: ${project.base-image.uri}: invalid reference format"} (Bad Request: 400)]
```

* If it's a feature request, what is your use case :

* Sample Reproducer Project : *[GitHub Clone URL]*

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,7 @@ Usage:
 * Fix #480: wildfly-jar doesn't depend on common-maven module
 * Fix #268: Generator and HealthCheck enrichers for Micronaut framework (JVM)
 * Fix #488: Controller enricher replica count can be set to `0` when ResourceConfig is provided
+* Fix #485: Filter with placeholders in Dockerfile is broken 
 
 ### 1.0.2 (2020-10-30)
 * Fix #429: Added quickstart for Micronaut framework

--- a/jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/helper/DockerFileUtil.java
+++ b/jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/helper/DockerFileUtil.java
@@ -63,7 +63,7 @@ public class DockerFileUtil {
      * @throws IOException if there's a problem while performin IO operations.
      */
     public static List<String> extractBaseImages(File dockerFile, Properties properties, String filter, Map<String, String> argsFromBuildConfig) throws IOException {
-        List<String[]> fromLines = extractLines(dockerFile, "FROM", properties, filter);
+        List<String[]> fromLines = extractLines(dockerFile, "FROM", properties, resolveDockerfileFilter(filter));
         Map<String, String> args = extractArgs(dockerFile, properties, filter, argsFromBuildConfig);
         Set<String> result = new LinkedHashSet<>();
         Set<String> fromAlias = new HashSet<>();
@@ -131,7 +131,7 @@ public class DockerFileUtil {
         try (BufferedReader reader = new BufferedReader(new FileReader(dockerFile))) {
             String line;
             while ((line = reader.readLine()) != null) {
-                ret.append(JKubeDockerfileInterpolator.interpolate(line, properties, filter)).append(System.lineSeparator());
+                ret.append(JKubeDockerfileInterpolator.interpolate(line, properties, resolveDockerfileFilter(filter))).append(System.lineSeparator());
             }
         }
         return ret.toString();
@@ -335,5 +335,9 @@ public class DockerFileUtil {
                 .filter(StringUtils::isNotBlank)
                 .forEach(ports::add));
         return new ArrayList<>(ports);
+    }
+
+    static String resolveDockerfileFilter(String filter) {
+        return filter != null ? filter : BuildConfiguration.DEFAULT_FILTER;
     }
 }

--- a/jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/helper/DockerFileUtilTest.java
+++ b/jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/helper/DockerFileUtilTest.java
@@ -104,6 +104,24 @@ public class DockerFileUtilTest {
     }
 
     @Test
+    public void interpolateWithNullFilterShouldPickDefaultFilter() throws IOException {
+        // Given
+        Properties properties = new Properties();
+        properties.put("project.base-image.uri", "openjdk:latest");
+        File givenDockerfile = new File(getClass().getResource("/interpolate/Dockerfile_with_params").getFile());
+
+        // When
+        String result = DockerFileUtil.interpolate(givenDockerfile, properties, null);
+
+        // Then
+        String[] lines = result.split("\n");
+        assertNotNull(result);
+        assertEquals(2, lines.length);
+        assertEquals("FROM openjdk:latest", lines[0]);
+        assertEquals("ENTRYPOINT [\"java\", \"-jar\", \"target/docker-file-simple.jar\"]", lines[1]);
+    }
+
+    @Test
     public void testCreateSimpleDockerfileConfig() throws IOException {
         // Given
         File dockerFile = File.createTempFile("Dockerfile", "-test");
@@ -258,6 +276,12 @@ public class DockerFileUtilTest {
         assertEquals(Arrays.asList("9090", "8080", "9999", "9010"), result1);
         assertEquals(Collections.singletonList("9001"), result2);
         assertEquals(Collections.singletonList("8001"), result3);
+    }
+
+    @Test
+    public void testResolveDockerfileFilter() {
+        assertEquals(BuildConfiguration.DEFAULT_FILTER, DockerFileUtil.resolveDockerfileFilter(null));
+        assertEquals("@*@", DockerFileUtil.resolveDockerfileFilter("@*@"));
     }
 
     private File getDockerfilePath(String dir) {

--- a/jkube-kit/build/api/src/test/resources/interpolate/Dockerfile_with_params
+++ b/jkube-kit/build/api/src/test/resources/interpolate/Dockerfile_with_params
@@ -1,0 +1,2 @@
+FROM ${project.base-image.uri}
+ENTRYPOINT ["java", "-jar", "target/docker-file-simple.jar"]

